packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,66 @@
 import Objection from 'objection';
+import {SignedState, calculateChannelId, Objective} from '@statechannels/wallet-core';
 
 import {SignState} from '../protocols/actions';
-import {Channel, SyncState} from '../models/channel';
+import {Channel, SyncState, RequiredColumns} from '../models/channel';
+import {SigningWallet} from '../models/signing-wallet';
+import {logger} from '../logger';
+import {addHash} from '../state-utils';
+import knex from '../db/connection';
 
 export const Store = {
+  transaction: async function(callback: () => Promise<void>): Promise<void> {
+    return knex.transaction(async () => {
+      await callback();
+    });
+  },
   signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
     const channel = await Channel.forId(action.channelId, tx);
     const notifications = channel.signAndAdd(action);
     await Channel.query(tx).update(channel);
 
     return notifications;
   },
+
+  addObjective: async function(_objective: Objective): Promise<void> {
+    // TODO: Implement this
+    return Promise.resolve();
+  },
+  addSignedState: async function(signedState: SignedState): Promise<void> {
+    if (!signedState.signatures?.length) {
+      throw new Error('No signature for signed state');
+    }
+
+    const channelId = calculateChannelId(signedState);
+    let channel = await Channel.query()
+      .where('channelId', channelId)
```

Comment:
There is a helper function in Andrew's PR (see [here](https://github.com/statechannels/statechannels/pull/2368/files#diff-ddee9e09686c5ac512163bd25f5122acR23-R28)) you could re-use